com.unity.netcode.gameobjects/Tests/Runtime/PlayerObjectTests.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Unity.Netcode.TestHelpers.Runtime;
+
+namespace Unity.Netcode.RuntimeTests
+{
+    [TestFixture(HostOrServer.Host)]
+    [TestFixture(HostOrServer.Server)]
+    public class PlayerObjectTests : NetcodeIntegrationTest
+    {
+        protected override int NumberOfClients => 1;
+
+        protected GameObject m_NewPlayerToSpawn;
+
+        public PlayerObjectTests(HostOrServer hostOrServer) : base(hostOrServer) { }
+
+        protected override void OnServerAndClientsCreated()
+        {
+            m_NewPlayerToSpawn = CreateNetworkObjectPrefab("NewPlayerInstance");
+            base.OnServerAndClientsCreated();
+        }
+
+        [UnityTest]
+        public IEnumerator SpawnAnotherPlayerObject()
+        {
+            var originalPlayer = m_PlayerNetworkObjects[1][1];
+            var playerLocalClient = m_ClientNetworkManagers[0].LocalClient.PlayerObject;
+            var newPlayer = Object.Instantiate(m_NewPlayerToSpawn);
+            var newPlayerNetworkObject = newPlayer.GetComponent<NetworkObject>();
+            newPlayerNetworkObject.NetworkManagerOwner = m_ServerNetworkManager;
+            newPlayerNetworkObject.SpawnAsPlayerObject(1);
+            yield return WaitForConditionOrTimeOut(() => !originalPlayer.IsPlayerObject && newPlayerNetworkObject.IsPlayerObject);
```

Comment:
Worth a comment for those who won't have this in their brain cache to say "Spawn another player object.  We should now see the original player object no longer think it's a player object",